packages/frontend-api/src/Model/Token/TokenFacade.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Token;
+
+use Lcobucci\JWT\Builder;
+use Lcobucci\JWT\Parser;
+use Lcobucci\JWT\Signer;
+use Lcobucci\JWT\Signer\Key;
+use Lcobucci\JWT\Token;
+use Lcobucci\JWT\ValidationData;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+use Symfony\Component\Security\Core\Exception\CustomUserMessageAuthenticationException;
+
+class TokenFacade
+{
+    protected const ACCESS_TOKEN_EXPIRATION = 3600;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Domain\Domain
+     */
+    protected $domain;
+
+    /**
+     * @var \Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface
+     */
+    protected $parameterBag;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface $parameterBag
+     */
+    public function __construct(
+        Domain $domain,
+        ParameterBagInterface $parameterBag
+    ) {
+        $this->domain = $domain;
+        $this->parameterBag = $parameterBag;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser $customerUser
+     * @return string
+     */
+    public function generateAccessTokenByCustomerUser(CustomerUser $customerUser): string
+    {
+        return $this->generateTokenByCustomerUserAndExpiration($customerUser, static::ACCESS_TOKEN_EXPIRATION);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser $customerUser
+     * @param int $expiration
+     * @return string
+     */
+    protected function generateTokenByCustomerUserAndExpiration(CustomerUser $customerUser, int $expiration): string
+    {
+        $currentTime = time();
+
+        $tokenBuilder = (new Builder())
+            ->issuedBy($this->domain->getUrl())
+            ->permittedFor($this->domain->getUrl())
+            ->issuedAt($currentTime)
+            ->expiresAt($currentTime + $expiration);
+
+        foreach (TokenCustomerUserTransformer::transform($customerUser) as $key => $value) {
+            $tokenBuilder->withClaim($key, $value);
+        }
+
+        return (string)$tokenBuilder->getToken($this->getSigner(), $this->getKey());
+    }
+
+    /**
+     * @return \Lcobucci\JWT\Signer\Key
+     */
+    public function getKey(): Key
+    {
+        return new Key($this->parameterBag->get('secret'));
+    }
+
+    /**
+     * @return \Lcobucci\JWT\Signer
+     */
+    public function getSigner(): Signer
+    {
+        return new Signer\Hmac\Sha256();
+    }
+
+    /**
+     * @param string $tokenString
+     * @return \Lcobucci\JWT\Token
+     */
+    public function getTokenByString(string $tokenString): Token
+    {
+        return (new Parser())->parse($tokenString);
```

Comment:
Hmm, I think you get 500 when you send an expired token as well.

But I do not know if exceptions should be caught here or at the place where this method will be called :thinking:

Reply:
The method is used several times in TokenAuthenticator. I would handle the exception here and throw UserError with some meaningful message ("token is not token at all" :D) instead of doing it every time I need to use the method.
But I don't have really strong preference about it.

Reply:
Hmm, I wrote about method `checkCredentials`. I don't know why. 8b5c43ac34426ebc0cd316016f355933b821573e